cardano/src/block/block.rs

```diff
@@ -287,13 +281,41 @@ impl cbor_event::de::Deserialize for BlockHeader {
                 let blk = cbor_event::de::Deserialize::deserialize(raw)?;
                 Ok(BlockHeader::MainBlockHeader(blk))
             },
-            _ => {
-                Err(cbor_event::Error::CustomError(format!("Unsupported BlockHeader: {}", sum_type_idx)))
+            idx => {
+                Err(cbor_event::Error::CustomError(format!("Unsupported BlockHeader: {}", idx)))
+            }
+        }
+    }
+}
+
+impl cbor_event::de::Deserialize for BlockHeaders {
+    fn deserialize<'a>(raw: &mut RawCbor<'a>) -> cbor_event::Result<Self> {
+        match decode_sum_type(raw)? {
+            0 => {
+                Ok(BlockHeaders(Vec::<BlockHeader>::deserialize(raw)?))
+            },
+            1 => {
+                Err(cbor_event::Error::CustomError(format!(
+                    "Server returned an error for Headers: {}",
+                    raw.text().unwrap())))
+            },
+            idx => {
+                Err(cbor_event::Error::CustomError(
+                    format!("Unsupported Headers: {}", idx)))
             }
         }
     }
 }
 
+fn decode_sum_type(raw: &mut RawCbor) -> cbor_event::Result<u64> {
```

Comment:
this function seems useful, could you push it down to `cardano/src/cbor/hs.rs` ?

Reply:
Done.